src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -418,28 +448,157 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
+        RegistryException
+    )
+
+    try:
+        request_data_from_registry(
+            http_method='get',
+            login_server=login_server,
+            path=_get_v2_manifest_path(repository_name, image),
+            username=username,
+            password=password,
+            json_payload=None,
+            file_payload=None,
+            manifest_headers=get_manifest_authorization_header(username, password),
+            params=image)
+
+    except RegistryException as e:
+        if e.status_code == 404:
+            _check_image_match(cmd,
+                               registry_name,
+                               repository_name,
+                               login_server,
+                               username,
+                               password,
+                               image)
+        raise CLIError("{} Please check if image/tag was inputted correctly.".format(str(e)))
+
+# Validate if blob layer can be pulled
+def _check_blob(repository_name,
+                login_server,
+                username,
+                password,
+                image):
+
+    import requests
+    from ._docker_utils import get_manifest_authorization_header, parse_error_message
+
+    # Get manifest
+    # GET {url}/v2/{name}/manifests/{reference}
+    manifest_url = 'https://{}/v2/{}/manifests/{}'.format(login_server, repository_name, image)
+
+    manifest_response = requests.get(
+        manifest_url,
+        headers=get_manifest_authorization_header(username, password))
+
+    manifest = manifest_response.json()
+
+    # Get the digest of the smallest blob for performance purposes
+    smallest_blob = min(manifest['layers'], key=lambda layer: layer['size'])
+    digest = smallest_blob['digest']
+
+    # Get blob
+    request_url = _get_blob(login_server, repository_name, digest)
+    logger.debug(add_timestamp("Sending a HTTP GET request to {}".format(request_url)))
+
+    res = requests.get(request_url, auth=(username, password))
+
+    # Return server error message; if server error not available, will return error message below
+    if res.status_code >= 400:
+        raise CLIError(parse_error_message('Could not get the requested data.', res), res.status_code)
+    # Check if image or tag was inputted and return appropriate message
+    if res.status_code < 400 and 'sha256' not in image:
+        print_pass("Blobs referenced by tag '{}' can be pulled from registry '{}'".format(image, login_server))
+    else:
+        print_pass("Blobs in image '{}' can be pulled from registry '{}'".format(image, login_server))
 
 # General command
 def acr_check_health(cmd,  # pylint: disable useless-return
                      vnet=None,
                      ignore_errors=False,
                      yes=False,
-                     registry_name=None):
+                     registry_name=None,
+                     repository_name=None,
+                     image=None,
+                     resource_group_name=None):
```

Comment:
isn't resource group by default available in all commands?

Reply:
also, could you please reorder the arguments by priority please?

Reply:
@rosanch - not certain about the resource group question, I thought I needed to include it in order to use in 'get_resource_group_by_registry_name' method on line 542?

Reply:
and yes can reorder the args

Reply:
I see you're correct. Thanks for clarifying